### ISSUE TITLE
incompatibility with Multiverse

### ISSUE THREAD

**Issue by woutercools** [synogen/ftlautosave#18]
It doesn't seem to work in conjunction with Multiverse en Hyperspace
https://subsetgames.com/forum/viewtopic.php?f=11&t=35332
https://subsetgames.com/forum/viewtopic.php?f=11&t=35095

For Hyperspace you have to downgrade FTL to version 1.6.9, maybe that is the cause.

I like ftlautosave a lot, but also Multiverse is excellent.
Would be great to be able to use both.

Thanks.

**Comment by changhe3** [comment]
No, the problem is much simpler: Hyperspace and MV uses `hs_mv_continue.sav` and `hs_mv_prof.sav` as save files... 
Just change them in ftlautosave.json

**Comment by woutercools** [comment]
> No, the problem is much simpler: Hyperspace and MV uses `hs_mv_continue.sav` and `hs_mv_prof.sav` as save files...
> Just change them in ftlautosave.json

Yes! That works! Thank you for your reply on this 1 year old post.

**Comment by woutercools** [comment]
Since Multiverse updated to version 5.2.3, FTLautosave can't read the gamesave files anymore:

![image](https://user-images.githubusercontent.com/16439116/163871288-83b559a6-258c-43f4-a290-26796d491cb2.png)

Discussion on the Multiverse Discord: https://discord.com/channels/604415384979898464/604418976277069855/965704762332971018

Can a new version accommodate for these changes?

**Comment by joerivandervelde** [comment]
I could take a look, but I don't have Multiverse. Could you provide the `hs_mv_continue.sav` and `hs_mv_prof.sav` files in question?

**Comment by woutercools** [comment]
> I could take a look, but I don't have Multiverse. Could you provide the `hs_mv_continue.sav` and `hs_mv_prof.sav` files in question?

[FasterThanLight.zip](https://github.com/synogen/ftlautosave/files/8523955/FasterThanLight.zip)

Here you go.

Thanks for your efforts.
Multiverse is an awesome mod: https://subsetgames.com/forum/viewtopic.php?t=35332

**Comment by joerivandervelde** [comment]
Thanks for the files. The save file format seems to be quite different from vanilla FTL saves, so AFAIK there is no easy fix unless perhaps a Multiverse developer steps in. Best I can do for now is to add a checkbox that lets you enable/disable the original strict parsing of FTL saves. For vanilla FTL saves, this makes no difference, although with strict parsing you can be more sure that the save is truly uncorrupted (cosmic ray bit flips? broken harddrive? dunno). For Multiverse saves, the strict parsing prevented you from getting save information, e.g.

```
Unexpected value 1600743278 at position 207
20 Apr 2022, 19:46:37 Save format version 11
    ! File could not be fully read, values might be missing or wrong !
  Ship null
    Hull: null    Fuel: null    Drone parts: null    Missiles: null    Scrap: null
  Stats
    Total ships defeated: null		Total crew obtained: null
    Total locations explored: null	Total scrap collected: null
```

But by disabling the strict parsing, the program continues to read what it can, resulting in the following information:

```
20 Apr 2022, 19:46:37 Save format version 11
  Ship Superluminal
    Hull: 1275592704    Fuel: 1    Drone parts: 0    Missiles: 1966080    Scrap: 786432
  Stats
    Total ships defeated: 0		Total crew obtained: 3
    Total locations explored: 1	Total scrap collected: 20
```

Obviously it's not perfect, some of the values are weirdly high. But better than nothing, I hope ;-) have a try and let me know what you think: https://github.com/joerivandervelde/ftlautosave/releases/tag/0.1.3-alpha

**Comment by woutercools** [comment]
Thanks for the fork!

A lot of gibberish numbers, but usable since you can identify the save by the amount of explored locations.

![image](https://user-images.githubusercontent.com/16439116/164759411-240b0c04-97ad-45e7-a1c9-52ba5c22e3e8.png)

**Comment by synogen** [comment]
I updated ftlautosave and made a new release 0.1.3. The save file parsing is less rigid by default now. Feel free to reopen if the error persists.

**Comment by woutercools** [comment]
I updated to Hyperspace 1.19.1 and Multiverse 5.5.1 but still get the "File could not be fully read" error.

![Image](https://github.com/user-attachments/assets/5f0c2897-bf3b-4ef9-b479-5d9e9b9c8df7)

**Comment by synogen** [comment]
Yep, as I said above, the parsing is less rigid now meaning it still reads whatever it can of the file even if it can't understand the format. The warning message is there to inform you that not all values could be read. So it's kind of like joerivandervelde's solution except that it stops reading once it notices the format is not as expected, so instead of random numbers you get a lot of zeroes and that warning.
I may or may not take a look at a Multiverse save file and see if I can adapt the format specification, but that's not a priority for me right now.

**Comment by synogen** [comment]
Turns out it was a relatively simple fix in the save format definition, so it works for my Multiverse saves now.
I made a new release 0.1.4 with that fix if you want to try it.

**Comment by woutercools** [comment]
It works correctly! Awesome!

Let's hope that a Hyperspace or Multiverse update doesn't break it again.

Thanks for letting us savescum in style again ;-)
